modules/cli/src/main/scala/scala/cli/commands/Package.scala

```diff
@@ -698,16 +707,25 @@ object Package extends ScalaCommand[PackageOptions] {
         val relSourceMapJs = os.rel / "main.js.map"
         val mainJs         = linkingDir / relMainJs
         val sourceMapJs    = linkingDir / relSourceMapJs
-        if (os.exists(mainJs)) {
-          os.copy(mainJs, dest, replaceExisting = true)
-          if (build.options.scalaJsOptions.emitSourceMaps && os.exists(sourceMapJs)) {
-            val sourceMapDest =
-              build.options.scalaJsOptions.sourceMapsDest.getOrElse(os.Path(s"$dest.map"))
-            os.copy(sourceMapJs, sourceMapDest, replaceExisting = true)
-            logger.message(s"Emitted js source maps to: $sourceMapDest")
+
+        if (os.exists(mainJs))
+          if (os.walk(linkingDir).filterNot(_ == mainJs).filterNot(_ == sourceMapJs).nonEmpty) {
```

Comment:
Let's use `os.walk.stream` here, that should allow not to list all files if ever there are a lot of them, or the directory contains garbage, etc.

```suggestion
          if (os.walk.stream(linkingDir).filterNot(_ == mainJs).filterNot(_ == sourceMapJs).nonEmpty) {
```
(if `geny.Generator` doesn't have a `nonEmpty` method, I guess `headOption.nonEmpty` should do…)